### PR TITLE
refactor(softdelete): Refactor soft delete

### PR DESCRIPTION
**Pull request by zhenlei520**
### Breaking Changes
* Isolation: The configuration is modified to
    ```
    {
      "ConnectionStrings": {
        "DefaultConnection": "server=localhost;uid=sa;pwd=P@ssw0rd;database=identity;"
      },
      "IsolationConnectionStrings": [
        {
          "TenantId": "*",//match all tenants
          "Environment": "development",
          "ConnectionString": "server=localhost,1434;uid=sa;pwd=P@ssw0rd;database=identity;",
          "Score": 99 //When multiple environments are matched according to the conditions, the highest one is selected as the link address of the current DbContext according to the descending order of scores. The default Score is 100.
        },
        {
          "TenantId": "00000000-0000-0000-0000-000000000002",
          "Environment": "development",
          "ConnectionString": "server=localhost,1435;uid=sa;pwd=P@ssw0rd;database=identity;"
        }
      ]
    }
    ```

### Features
* Data.Contracts: Add `ISoftDelete`、`IDataFilter`
* Data: Add Masa.BuildingBlocks.Data

### Docs:
* Isolation.UoW.EF: Modify the Readme document

### Refactors
* Ddd.Domain: `IAuditEntity` supports soft delete
* Data.UoW: `IDbConnectionStringProvider` moved to Masa.BuildingBlocks.Data
* Isolation: Delete `IIsolationDbConnectionStringProvider`

### CODE REVIEW

File: src/Data/Masa.BuildingBlocks.Data/ConnectionStringNameAttribute.cs [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    public static string GetConnStringName<T>() => GetConnStringName(typeof(T));
+
+    public static string GetConnStringName(Type type)
+    {
+        var nameAttribute = type.GetTypeInfo().GetCustomAttribute<ConnectionStringNameAttribute>();
+
+        if (nameAttribute == null)
+            return ConnectionStrings.DEFAULT_CONNECTION_STRING_NAME;
+
+        return !string.IsNullOrEmpty(nameAttribute.Name)? nameAttribute.Name : type.FullName!;
```
Comment (doddgu) [inline src/Data/Masa.BuildingBlocks.Data/ConnectionStringNameAttribute.cs:22]: code format

File: src/Data/Masa.BuildingBlocks.Data/IConnectionStringProvider.cs [hunk 2]
```diff
@@ -0,0 +1,21 @@
+﻿// Copyright (c) MASA Stack All rights reserved.
+// Licensed under the MIT License. See LICENSE.txt in the project root for license information.
+
+namespace Masa.BuildingBlocks.Data;
+
+public interface IConnectionStringProvider
+{
+    /// <summary>
+    /// Get database link string based on ConnectionName
```
Comment (doddgu) [inline src/Data/Masa.BuildingBlocks.Data/IConnectionStringProvider.cs:9]: link string?

**Review by doddgu — CHANGES_REQUESTED** [review]

File: src/Data/Masa.BuildingBlocks.Data/IConnectionStringProvider.cs (hunk 2 above)
Comment (zhenlei520) [reply src/Data/Masa.BuildingBlocks.Data/IConnectionStringProvider.cs]: processed

File: src/Data/Masa.BuildingBlocks.Data/ConnectionStringNameAttribute.cs (hunk 1 above)
Comment (zhenlei520) [reply src/Data/Masa.BuildingBlocks.Data/ConnectionStringNameAttribute.cs]: processed

**Review by doddgu — APPROVED** [review]
LGTM

**Pull request merged**